fw/cfg.h

```diff
@@ -194,26 +194,71 @@ typedef struct {
 	     (idx)++,				\
 	     (v) = (idx < (e)->val_n ? (e)->vals[(idx)] : NULL))
 
-#define TFW_CFG_CHECK_NO_ATTRS(spec, entry)				\
-	if ((entry)->attr_n) {						\
-		T_ERR_NL("%s: Arguments may not have the '=' sign\n",	\
-			 (spec)->name);					\
-		return -EINVAL;						\
-	}
+#define TFW_CFG_CHECK_NO_ATTRS(spec, entry)			\
+	do {							\
+		if ((entry)->attr_n) {				\
+			T_ERR_NL("%s: Arguments may not have "	\
+				"the '=' sign\n", (spec)->name);\
+			return -EINVAL;				\
+		}						\
+	} while (0)
+
+#define TFW_CFG_CHECK_ATTR_N(op, req, spec, entry)			\
+	do {								\
+		if (!((entry)->attr_n op (req))) {			\
+			T_ERR_NL("%s: Invalid number of attributes: "	\
+				"%zu, must be %s %d\n",			\
+				(spec)->name, (entry)->attr_n,		\
+				#op, (req));				\
+			return -EINVAL;					\
+		}							\
+	} while (0)
+
+#define TFW_CFG_CHECK_ATTR_EQ_N(req, spec, entry)			\
+	do {								\
+		if (!((entry)->attr_n == (req))) {			\
+			T_ERR_NL("%s: Invalid number of attributes: "	\
+				"%zu, must be queal %d\n",		\
+				(spec)->name, (entry)->attr_n, (req));	\
+			return -EINVAL;					\
+		}							\
+	} while (0)
+
+#define TFW_CFG_CHECK_ATTR_LE_N(req, spec, entry)			\
```

Comment:
I had to add such macro because old versions with op as the first arg are no passed by linter, because it requires spaces around op values such as ==